infra/pipelines/pipelines/order-generator.ts

```diff
@@ -0,0 +1,247 @@
+      }],
+      stages: [
+        {
+          name: "Github",
+          actions: [{
+              name: "Github",
+              category: "Source",
+              owner: "AWS",
+              provider: "CodeStarSourceConnection",
+              version: "1",
+              outputArtifacts: ["source_output"],
+              configuration: {
+                  ConnectionArn: connection.arn,
+                  FullRepositoryId: "boundless-xyz/boundless",
+                  BranchName: BRANCH_NAME,
+                  OutputArtifactFormat: "CODEBUILD_CLONE_REF"
+              },
+          }],
+        },
+        {
+          name: "DeployStaging",
+          actions: [
+            {
+              name: "DeployStaging",
+              category: "Build",
+              owner: "AWS",
+              provider: "CodeBuild",
+              version: "1",
+              runOrder: 1,
+              configuration: {
+                ProjectName: stagingDeployment.name
+              },
+              outputArtifacts: ["staging_output"],
+              inputArtifacts: ["source_output"],
+            }
+          ]
+        },
+        {
+          name: "DeployProduction",
+          actions: [
+            { name: "ApproveDeployToProduction", 
+              category: "Approval", 
+              owner: "AWS", 
+              provider: "Manual", 
+              version: "1", 
+              runOrder: 1,
+              configuration: {}
+            },
+            {
+              name: "DeployProduction",
+              category: "Build",
+              owner: "AWS",
+              provider: "CodeBuild",
+              version: "1",
+              runOrder: 2,
+              configuration: {
+                ProjectName: prodDeployment.name
+              },
+              outputArtifacts: ["production_output"],
+              inputArtifacts: ["source_output"],
+            }
+          ]
+        }
+      ],
+      triggers: [
+        {
+          providerType: "CodeStarSourceConnection",
+          gitConfiguration: {
+            sourceActionName: "Github",
+            pushes: [
+              {
+                branches: {
+                  includes: [BRANCH_NAME],
+                },
+              },
+            ],
+          },
+        },
+      ],
+      name: `${APP_NAME}-pipeline`,
+      roleArn: role.arn,
+    });
+  }
+
+  private codeBuildProjectArgs(
+    appName: string, 
+    stackName: string, 
+    role: aws.iam.Role, 
+    serviceAccountRoleArn: string, 
+    dockerUsername: string, 
+    dockerTokenSecret: aws.secretsmanager.Secret, 
+    githubTokenSecret: aws.secretsmanager.Secret
+  ): aws.codebuild.ProjectArgs {
+    return {
+      buildTimeout: 60,
+      description: `Deployment for ${APP_NAME}`,
+      serviceRole: role.arn,
+      environment: {
+        computeType: "BUILD_GENERAL1_MEDIUM",
+        image: "aws/codebuild/standard:7.0",
+        type: "LINUX_CONTAINER",
+        privilegedMode: true,
```

Comment:
does it needed?

Reply:
`privilegedMode`? Yes it is needed for doing dockerbuilds in CodeBuild